src/ipc-dbus/DBusIPCAPI_v1.cpp

```diff
@@ -1267,6 +1269,8 @@ DBusIPCAPI_v1::interface_config_gateway_handler(
 		DBUS_TYPE_BOOLEAN, &configure,
 		DBUS_TYPE_BOOLEAN, &stable,
 		DBUS_TYPE_UINT16, &prefix_len_in_bits,
+		DBUS_TYPE_BOOLEAN, &nd_dns,
+		DBUS_TYPE_BOOLEAN, &domain_prefix,
 		DBUS_TYPE_INVALID
 	);
 
```

Comment:
@XPAWAN , follow up from https://github.com/openthread/wpantund/pull/482#discussion_r588782383, we have two ways to fix 
1) Update the `tool-cmd-remove-prefix` to also add the `nd_ds` and `domain_prefix` flags in DBus message it sends
2) Make sure the parser here handle both cases (with/without extra flags)

The second option is much better, since it keeps the DBus API of wpantund backward-compatible.

We generally aim to keep the DBus API backward compatible (unless there is a good justification/reason for it). This is to ensure newer versions of `wpantund` continues to work with older modules/processes that may interact with `wpantund` through its DBus API.

Here is my suggestion to add this here:

```
	if (!did_succeed) {
		did_succeed = dbus_message_get_args(
			message, NULL,
			DBUS_TYPE_BOOLEAN, &default_route,
			DBUS_TYPE_ARRAY, DBUS_TYPE_BYTE, &prefix, &prefix_len_in_bytes,
			DBUS_TYPE_UINT32, &preferred_lifetime,
			DBUS_TYPE_UINT32, &valid_lifetime,
			DBUS_TYPE_BOOLEAN, &preferred,
			DBUS_TYPE_BOOLEAN, &slaac,
			DBUS_TYPE_BOOLEAN, &on_mesh,
			DBUS_TYPE_INT16, &priority_raw,
			DBUS_TYPE_BOOLEAN, &dhcp,
			DBUS_TYPE_BOOLEAN, &configure,
			DBUS_TYPE_BOOLEAN, &stable,
			DBUS_TYPE_UINT16, &prefix_len_in_bits,
			DBUS_TYPE_INVALID
		);
	}
```

I think/expect with such a change here and https://github.com/openthread/openthread/pull/6240 in OT, the CI tests should all pass.

Reply:
Hello @abtink. Thank You for Your support. 
I have added a new commit according to Your remarks.
The OT related PR is already merged( openthread/openthread#6174 )
Shall I also create a new official PR with changes for openthread as in  https://github.com/openthread/openthread/pull/6240 ?

Reply:
I just re-ran the test and it did indeed pass ok 
- https://github.com/openthread/openthread/pull/6240/checks?check_run_id=2058442009

--------

> Shall I also create a new official PR with changes for openthread as in openthread/openthread#6240 ?

I will submit it as PR now (removing the top commit with fixed it to use the `wpantund` from this branch).

Here is the plan:
- We merge the https://github.com/openthread/openthread/pull/6240 first.
- After that, we can re-run  CI on this PR (which should hopefully then pass).
- Then we can then merge this PR.